sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1255,6 +1255,30 @@ class DataFrame private[sql](
   }
 
   /**
+   * Returns a new [[DataFrame]] with columns dropped.
+   * This is a no-op if schema doesn't contain column name(s).
+   * @group dfops
+   * @since 1.6.0
+   */
+  def drop(colNames: String*): DataFrame = {
```

Comment:
This would need a varargs annotation and I don't think that we should duplicate the column resolution logic.  Otherwise it might fall out of sync.